webapp/src/components/expanded_view/call_settings.tsx

```diff
@@ -0,0 +1,477 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
+import CCIcon from 'src/components/icons/cc_icon';
+import SettingsWheelIcon from 'src/components/icons/settings_wheel';
+import ShowMoreIcon from 'src/components/icons/show_more';
+import SpeakerIcon from 'src/components/icons/speaker_icon';
+import TickIcon from 'src/components/icons/tick';
+import UnmutedIcon from 'src/components/icons/unmuted_icon';
+import {areLiveCaptionsAvailableInCurrentCall} from 'src/selectors';
+import type {
+    AudioDevices,
+} from 'src/types/types';
+import {getCallsClient} from 'src/utils';
+import styled from 'styled-components';
+
+import ControlsButton from './controls_button';
+
+type AudioDevicesListProps = {
+    deviceType: string;
+    devices: MediaDeviceInfo[];
+    currentDevice: MediaDeviceInfo | null;
+    onDeviceClick: (device: MediaDeviceInfo) => void;
+}
+
+const AudioDevicesList = ({deviceType, devices, currentDevice, onDeviceClick}: AudioDevicesListProps) => {
+    const {formatMessage} = useIntl();
+
+    // Note: this is system default, not the concept of default that we save in local storage in client.ts
+    const makeDeviceLabel = (device: MediaDeviceInfo) => {
+        if (device.deviceId.startsWith('default') && !device.label.startsWith('Default')) {
+            return formatMessage({defaultMessage: 'Default - {deviceLabel}'}, {deviceLabel: device.label});
+        }
+        return device.label;
+    };
+
+    const list = devices.map((device) => {
+        const isCurrentDevice = device.deviceId === currentDevice?.deviceId;
+        return (
+            <li
+                className='MenuItem'
+                key={`audio-${deviceType}-device-${device.deviceId}`}
+            >
+                <AudioDeviceButton
+                    className='style--none'
+                    onClick={() => onDeviceClick(device)}
+                    $isCurrentDevice={isCurrentDevice}
+                >
+                    <AudioDeviceName>
+                        {makeDeviceLabel(device)}
+                    </AudioDeviceName>
+                    { isCurrentDevice &&
+                    <AudioDeviceSelectedIcon>
+                        <TickIcon/>
+                    </AudioDeviceSelectedIcon>
+                    }
+                </AudioDeviceButton>
+            </li>
+        );
+    });
+
+    return (
+        <div className='Menu'>
+            <DevicesList
+                id={`calls-popout-audio-${deviceType}s-menu`}
+                className='Menu__content dropdown-menu'
+            >
+                {list}
+            </DevicesList>
+        </div>
+    );
+};
+
+const AudioDeviceSelectedIcon = styled.div`
+&&&& {
```

Comment:
For this and most of the others, are you sure we need this much specificity? I don't think I've needed to add that in awhile...

For the svg on the next line, a `> svg` should be enough?

Reply:
I really struggled with this because most of them actually needed the extra specificity due to how our menu classes are defined. And that's why in-line styling was working so well, but someone introduced styled components :p

Reply:
I can review, it just takes time.